node/ChainwebNode.hs

```diff
@@ -338,22 +338,29 @@ node conf logger = do
     rocksDbDir <- getRocksDbDir conf
     pactDbDir <- getPactDbDir conf
     dbBackupsDir <- getBackupsDir conf
-    (if _configOnlySyncPact cwConf then withReadOnlyRocksDb else withRocksDb)
-        rocksDbDir modernDefaultOptions $ \rocksDb -> do
-            logFunctionText logger Info $ "opened rocksdb in directory " <> sshow rocksDbDir
-            logFunctionText logger Info $ "backup config: " <> sshow (_configBackup cwConf)
-            withChainweb cwConf logger rocksDb pactDbDir dbBackupsDir (_nodeConfigResetChainDbs conf) $ \case
-                Replayed _ _ -> return ()
-                StartedChainweb cw ->
-                    concurrentlies_
-                        [ runChainweb cw
-                        -- we should probably push 'onReady' deeper here but this should be ok
-                        , runCutMonitor (_chainwebLogger cw) (_cutResCutDb $ _chainwebCutResources cw)
-                        , runQueueMonitor (_chainwebLogger cw) (_cutResCutDb $ _chainwebCutResources cw)
-                        , runRtsMonitor (_chainwebLogger cw)
-                        , runBlockUpdateMonitor (_chainwebLogger cw) (_cutResCutDb $ _chainwebCutResources cw)
-                        , runDatabaseMonitor (_chainwebLogger cw) rocksDbDir pactDbDir
-                        ]
+    withRocksDb' <-
+        if _configOnlySyncPact cwConf
+        then
+            if _cutPruneChainDatabase (_configCuts cwConf) == GcNone
+            then withReadOnlyRocksDb <$ logFunctionText logger Info "Opening RocksDB in read-only mode"
+            else withRocksDb <$ logFunctionText logger Info "Opening RocksDB in read-write mode, if this wasn't intended, ensure that cuts.pruneChainDatabase is set to none"
```

Comment:
For a follow-up PR:

I think we should just move the existing pruning functionality into `cwtool`.

Merkle-tree validation should be inlined with pact replay. This is the default use case for endusers who need to initialize a node from a untrusted rocksdb. We should carefully check the code whether additional checks would have to be added to replay. For payload the current behavior (with payload hash validation enabled) should be sufficient. So, we would have to add header validation (including braiding validation), which is generally fast and should create only minimal overhead.

For other use cases the existing functionality could be used via `cwtool`.